Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp

```diff
@@ -469,6 +484,12 @@ namespace PhysX
     void ArticulationLinkComponent::DestroyArticulation()
     {
         AzPhysics::Scene* scene = AZ::Interface<AzPhysics::SceneInterface>::Get()->GetScene(m_attachedSceneHandle);
+        if (scene == nullptr)
+        {
+            // The scene can be removed before articulation is destroyed.
+            // If the scene was removed. Articulations were also removed.
+            return;
+        }
```

Comment:
does `m_articulationLinks` still get cleared when the scene is removed?

Reply:
Yeah this check should be scoped to be only around the scene and pxScene statements.

Reply:
This check prevents the function from running twice. If the scene was removed this function would run due to the handler and after the component will be deactivated, not running the `DestroyArticulation` function (as the scene at that point would be nullptr). If the scene wasn't removed the handler gets disconnected and again this function runs normally.